src/tx/tx.service.ts

```diff
@@ -88,6 +84,10 @@ export class TxService {
   }
 
   private async createTxWithAction(nonce: bigint, action: Value): Promise<Tx> {
+    const genesisHash = Buffer.from(
+      this.configService.getOrThrow<string>('GENESIS_BLOCK_HASH'),
+      'hex',
+    );
```

Comment:
It is just my favor but how about moving this part to the `constructor` method? Then it may be failed to construct `TxService` and the application will not run (early fail) (but I'm not sure for the NestJS lifetime)

Reply:
`https://docs.nestjs.com/techniques/configuration#schema-validation`
We can use joi at this time for validate every env at time.